aks/application/data.tf

```diff
@@ -0,0 +1,19 @@
+data "azurerm_resource_group" "monitoring" {
+  count = var.azure_enable_container_monitoring ? 1 : 0
+
+  name = "${var.azure_resource_prefix}-${var.service_short}-mn-rg"
+}
+
+data "azurerm_monitor_action_group" "main" {
+  count = var.azure_enable_container_monitoring ? 1 : 0
+
+  name                = "${var.azure_resource_prefix}-${var.service_name}"
+  resource_group_name = data.azurerm_resource_group.monitoring[0].name
+}
+
+data "azurerm_kubernetes_cluster" "main" {
+  count = var.azure_enable_container_monitoring ? 1 : 0
+
+  name                = "${var.cluster_configuration_map.resource_prefix}-aks"
+  resource_group_name = var.cluster_configuration_map.resource_group_name
+}
```

Comment:
Could we expose the `id` of the cluster as an output from the `cluster_data` module to avoid needing these duplicate lines here. Since we've only got these lines here to fetch the `id` and we already have these lines in the cluster data module:

https://github.com/DFE-Digital/terraform-modules/blob/83801213853ed1e4b4bdcb8d36773c8683ff010f/aks/cluster_data/data.tf#L1-L4